bin/segment2.c

```diff
@@ -1900,239 +1460,868 @@ ALIGNED8 static const Texture texture_hud_char_arrow_down[] = {
+#include "levels/menu/main_menu_seg7.083D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_ko[] = {
+#include "levels/menu/main_menu_seg7.085D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_kana_handakuten_pi[] = {
+#include "levels/menu/main_menu_seg7.087D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_long_vowel[] = {
+#include "levels/menu/main_menu_seg7.089D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_hiragana_su[] = {
+#include "levels/menu/main_menu_seg7.08BD0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_hiragana_ru[] = {
+#include "levels/menu/main_menu_seg7.08DD0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_hiragana_ke[] = {
+#include "levels/menu/main_menu_seg7.08FD0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_ma[] = {
+#include "levels/menu/main_menu_seg7.091D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_ri[] = {
+#include "levels/menu/main_menu_seg7.093D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_o[] = {
+#include "levels/menu/main_menu_seg7.095D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_su[] = {
+#include "levels/menu/main_menu_seg7.097D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_a[] = {
+#include "levels/menu/main_menu_seg7.099D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_hiragana_mi[] = {
+#include "levels/menu/main_menu_seg7.09BD0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_hira_dakuten_do[] = {
+#include "levels/menu/main_menu_seg7.09DD0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_hiragana_no[] = {
+#include "levels/menu/main_menu_seg7.09FD0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_sa[] = {
+#include "levels/menu/main_menu_seg7.0A3D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_u[] = {
+#include "levels/menu/main_menu_seg7.0A5D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_n[] = {
+#include "levels/menu/main_menu_seg7.0A7D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_kana_dakuten_do[] = {
+#include "levels/menu/main_menu_seg7.0A9D0.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_katakana_ra[] = {
+#include "levels/menu/main_menu_seg7.katakana_ra.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_kana_dakuten_ge[] = {
+#include "levels/menu/main_menu_seg7.kana_dakuten_ge.rgba16.inc.c"
+};
+
+ALIGNED8 static const Texture texture_hud_char_kana_dakuten_ji[] = {
+#include "levels/menu/main_menu_seg7.kana_dakuten_ji.rgba16.inc.c"
+};
+#endif
+
+// ASCII lookup table for the colorful HUD font
+const struct AsciiCharLUTEntry main_hud_lut[] = {
```

Comment:
whats the point? this table is specifically for rendered characters. custom commands below 32 can be handled in a switch case just fine, this table is purely for rendering glyphs and handling kernings in the default case. if someone wants to actually print a normal character that isn't an ascii one they should use a UTF8 character instead

Reply:
I specify "commands" because the current system in place doesn't work well with multi-byte characters. Example of what I mean is `@` being used for color, only maybe somebody wouldn't want to use a real ASCII character to do that. Regarding the render table, since I see `c - ' '` being used all over the place, I'm worried that using a smaller index character could lead to UB somewhere.

Reply:
yeah but with the current system @ could easily be replaced with say \x01 without issues because commands are checked before this table is. and if you used a smaller index it would just render a garbage character, same as if you tried to use a nonmapped character in vanilla. i don't see a reason to add an extra kilobyte (32 entries * 8 bytes * 4 tables) of useless data just for this

Reply:
Would it render garbage or could it crash? You're storing a pointer, after all.

Reply:
If you're certain that there won't be any UB happening with proper command checks skipping the default character render, then I'll drop the issue.